plugin.video.orftvthek/resources/lib/htmlscraper.py

```diff
@@ -0,0 +1,1040 @@
+				inputstreamAdaptive = False
+
+			if channel:
+				channel = "[%s]" % channel
+			else:
+				channel = "LIVE"
+
+			uhd_streaming_url = self.getLivestreamUrl(data, 'uhdbrowser')
+			if uhd_streaming_url:
+				debugLog("Adding UHD Livestream")
+				uhdContextMenuItems = []
+				if inputstreamAdaptive and restart and online:
+					uhdContextMenuItems.append(('Restart', 'RunPlugin(plugin://%s/?mode=liveStreamRestart&link=%s)' % (
+						xbmcaddon.Addon().getAddonInfo('id'), link)))
+					uhd_final_title = "[Restart]%s[UHD] - %s%s" % (channel, title, time_str)
+				else:
+					uhd_final_title = "%s[UHD] - %s%s" % (channel, title, time_str)
+				self.html2ListItem(uhd_final_title, banner, "", state, time, channel, channel, generateAddonVideoUrl(uhd_streaming_url), None, False, True, uhdContextMenuItems)
+
+			streaming_url = self.getLivestreamUrl(data, self.videoQuality)
+			contextMenuItems = []
+			if inputstreamAdaptive and restart and online:
+				contextMenuItems.append(('Restart', 'RunPlugin(plugin://%s/?mode=liveStreamRestart&link=%s)' % (
+					xbmcaddon.Addon().getAddonInfo('id'), link)))
+				final_title = "[Restart]%s - %s%s" % (channel, title, time_str)
+			else:
+				final_title = "%s - %s%s" % (channel, title, time_str)
+			self.html2ListItem(final_title, banner, "", state, time, channel, channel, generateAddonVideoUrl(streaming_url), None, False, True, contextMenuItems)
+
+	def liveStreamRestart(self, link):
+		try:
+			xbmcaddon.Addon('inputstream.adaptive')
+		except RuntimeError:
+			return
+
+		html = fetchPage({'link': link})
+		bitmovinStreamId = self.getLivestreamBitmovinID(html)
+		stream_info = self.getLivestreamInformation(html)
+
+		if bitmovinStreamId:
+			title = stream_info['title']
+			image = stream_info['image']
+			description = stream_info['description']
+			duration = stream_info['duration']
+			date = stream_info['date']
+			channel = stream_info['channel']
+
+			ApiKey = '2e9f11608ede41f1826488f1e23c4a8d'
+			response = url_get_request('https://playerapi-restarttv.ors.at/livestreams/%s/sections/?state=active&X-Api-Key=%s' % (bitmovinStreamId, ApiKey))
+			try:
+				charset = response.headers.get_content_charset()
+				response_raw = response.read().decode(charset)
+			except AttributeError:
+				response_raw = response.read().decode('utf-8')
+
+			section = json.loads(response_raw)
+			if len(section):
+				section = section[0]
+				streamingURL = 'https://playerapi-restarttv.ors.at/livestreams/%s/sections/%s/manifests/hls/?startTime=%s&X-Api-Key=%s' % (bitmovinStreamId, section.get('id'), section.get('metaData').get('timestamp'), ApiKey)
+
+				listItem = createListItem(title, image, description, duration, date, channel, streamingURL, True, False, self.defaultbackdrop, self.pluginhandle)
+				listItem.setProperty('inputstreamaddon', 'inputstream.adaptive')
+				listItem.setProperty('inputstream.adaptive.manifest_type', 'hls')
+				self.xbmc.Player().play(streamingURL, listItem)
+
+	@staticmethod
```

Comment:
it'd be nice if you can pass your addon through reindent: https://pypi.org/project/reindent/
Your identation in this file is a nightmare :) PEP recommends 4 spaces.

Reply:
sorry for the mess. i ran reindent on all py files.